### ISSUE TITLE
Queue time on batched hits?

### ISSUE THREAD

**Issue by mikkelrj** [brsanthu/google-analytics-java#51]
After some experimentation with batching enabled, it seems that all hits come in at the same time point (the time of the batch request), when I really want the hits to appear at the time they were posted. This seems like a general issue with using batched requests on GA, but I have been unable to find anything about it. Any pointers would be appreciated.

Anyway, as a solution, I am considering adding the _Queue Time_ parameter to each hit, but I can see no other way than deriving the qt value at the time when the batch request is made, which means it has to be done when building the batch request in the `GoogleAnalyticsImpl.submitBatch(boolean)` method. Only then is the time offset of each hit known. This requires keeping time of the batched hits.

Any thoughts or maybe consideration of implementing the option of keeping time of the hits and setting the queue time parameter automatically, when in batch mode, at the time of posting the batch request? 

Thanks!
---
Documentation on the Queue Time parameter: https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#qt

**Comment by brsanthu** [comment]
We will need to add a submitTime to `GoogleAnalyticsRequest` to capture time when it was submitted by the consumer. At the time of posting to GA, we need to calculate the queue time.

The issue I see here is, if a customer has already using the queue time parameter, then it will be a conflict. We could populate the queue time if it is not already populated but that will still leave the issue as is for those who are using queue time and want to batch.

**Comment by mikkelrj** [comment]
Right. And I see the possible conflict with existing use of the queue time parameter. 

I was thinking it could make sense to add a property to `GoogleAnalyticsConfig` (maybe `autoQueueTime`?) to explicitly control this behaviour. When this property is `true` and `batchingEnabled` is `true` then the queue time property would be calculated and set on each request at the time the batch request is posted to GA.

**Comment by brsanthu** [comment]
That makes sense. Since `qt` value is a milliseconds, if customers enables it (we could flag as defaults to true), then we will add to the existing value if there is one or set new.

**Comment by mikkelrj** [comment]
I can maybe implement this in the weekend, unless this is something you plan to do.

**Comment by brsanthu** [comment]
Added `setAutoQueueTime` config flag. If enabled, it adds or sets the queue time from time request created to request submitted.
